### PR TITLE
refactor(ops): op2 supports strings in argument position

### PR DESCRIPTION
**Pull request by mmastrac**
Waiting for https://github.com/denoland/rusty_v8/pull/1261

### CODE REVIEW

File: core/runtime/ops.rs [hunk 1]
```diff
@@ -197,6 +200,204 @@ pub fn to_i64(number: &v8::Value) -> i32 {
...
+    debug_assert!(written <= capacity);
+    // We know it's valid UTF-8, so make a string
+    String::from_raw_parts(out, written, capacity)
+  }
+}
+
+/// Converts a [`v8::String`] to either an owned string, or a borrowed string, depending on whether it fits into the
+/// provided buffer.
+#[inline(always)]
+pub fn to_str<'a, const N: usize>(
```
Comment (mmastrac) [inline core/runtime/ops.rs:286]: This will go away with https://github.com/denoland/rusty_v8/pull/1261

File: core/runtime/ops.rs [hunk 2]
```diff
@@ -406,6 +608,127 @@ mod tests {
...
+      "op_test_string_option_return",
+      "assert(op_test_string_option_return('none', 'b') == null)",
+    )?;
+    run_test2(
+      1,
+      "op_test_string_roundtrip",
+      "assert(op_test_string_roundtrip('\\u0080\\u00a0\\u00ff') == '\\u0080\\u00a0\\u00ff')",
+    )?;
+    Ok(())
+  }
```
Comment (bartlomieju) [inline core/runtime/ops.rs:730]: Great tests 👍

File: core/runtime/ops.rs (hunk 1 above)
Comment (bartlomieju) [reply core/runtime/ops.rs:286]: rusty_v8 has been upgraded to v0.74.1 so you can remove it now

File: ops/op2/dispatch_slow.rs [hunk 3]
```diff
@@ -198,13 +203,31 @@ pub fn from_arg(
...
       quote! {
-        let #arg_ident = #arg_ident.to_rust_string_lossy();
+        let #arg_ident = #arg_ident.to_rust_string_lossy(#scope);
       }
     }
     Arg::Special(Special::RefStr) => {
+      *needs_scope = true;
+      quote! {
+        // Trade 1024 bytes of stack space for potentially non-allocating strings
+        let mut #arg_temp: [::std::mem::MaybeUninit<u8>; 1024] = [::std::mem::MaybeUninit::uninit(); 1024];
```
Comment (bartlomieju) [inline ops/op2/dispatch_slow.rs:221]: Should we keep it at 1024 or maybe go with something bigger like 8Kb?

**Review by bartlomieju — APPROVED** [review]
LGTM, great progress

File: ops/op2/dispatch_slow.rs (hunk 3 above)
Comment (mmastrac) [reply ops/op2/dispatch_slow.rs:221]: Let's leave it at 1024 for now (which should cover a pretty big chunk basic strings) and then later on we can experiment with the right size with some test cases to figure out the right balance.

**Pull request merged**